spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/datasource/PartitioningUtils.scala

```diff
@@ -78,7 +78,12 @@ object PartitioningUtils {
     MessageFormat.format(partitioningPattern, reportYear, reportMonth, reportDay)
   }
 
-  private def validateReportDate(reportDate: String): Unit = {
+  /**
+   * Validates the port date to be in tyhe expected format: 'yyyy-MM-dd'.
+   *
+   * @param reportDate          A report date string
+   */
+  def validateReportDate(reportDate: String): Unit = {
     val reportDateRegEx = """\d\d\d\d-\d\d-\d\d""".r
 
     if (!reportDateRegEx.pattern.matcher(reportDate).matches()) {
```

Comment:
```suggestion
    reportDate match {
      case reportDateRegEx() =>
      case _ => throw new IllegalArgumentException("...")
    }
```

Also, maybe it would be useful if the regex was outside of the method, so it doesn't need to be recreated every time. it is used. 

Either private, if planned to be used only here and in this way, or maybe as `"""(\d{4})-(\d{2})-(\d{2})"""` with capturing groups if reused from somewhere else (in that case, the matching would have to change to ```reportDateRegEx(y, m, d)```).

Reply:
Sure, thanks. Will fix it since I'm there. But the piece of code you added the comment to is not part of this PR.

Reply:
I know, sorry about that 😈

Reply:
Nw, it is a good suggestion 😄

Reply:
Actually, now that I'm looking at it, I prefer the current implementation. The matcher implementation has an empty case. But this opinion is not strong. If people prefer it as a matcher, I can change.

Reply:
Unless it will be used to extract those parts too, I think it would not make much of a difference